app/src/state/parse-settings.ts

```
@@ -308,6 +319,65 @@ function generateSettingsXml(stateSettings: ChartSettings, indent: string): stri
   xml += `${indent}<setting name="enHideBlockLabel">${stateSettings.enableHideBlockLabel}</setting>\n`;
   return xml;
 }
+
+export function translateColumnInfoToColumnInfoTSC(state: ColumnInfo): ColumnInfoTSC {
+  let column: ColumnInfoTSC = JSON.parse(JSON.stringify(defaultColumnBasicInfoTSC));
+  switch (state.columnDisplayType) {
+    case "Event":
+      //can't set it equal to default because it becomes reference to object
+      column = JSON.parse(JSON.stringify(defaultEventColumnInfoTSC));
+      break;
+    case "Zone":
+      column = JSON.parse(JSON.stringify(defaultZoneColumnInfoTSC));
+      break;
+    case "Sequence":
+      column = JSON.parse(JSON.stringify(defaultSequenceColumnInfoTSC));
+      break;
+    case "Range":
+      column = JSON.parse(JSON.stringify(defaultRangeColumnInfoTSC));
+      break;
+    case "Ruler":
+      column = JSON.parse(JSON.stringify(defaultRulerColumnInfoTSC));
+      break;
+    case "Point":
+      column = JSON.parse(JSON.stringify(defaultPointColumnInfoTSC));
+  }
+  switch (state.columnDisplayType) {
+    case "RootColumn":
+    case "MetaColumn":
+    case "BlockSeriesMetaColumn":
+      column._id = `class datastore.${state.columnDisplayType}:` + state.name;
+      break;
+    default:
+      column._id = `class datastore.${state.columnDisplayType}Column:` + state.name;
+  }
+  column.title = escapeHtmlChars(state.editName, "text");
+  column.isSelected = state.on;
+  column.drawTitle = state.enableTitle;
+  column.fonts = state.fontsInfo;
+  if (state.width) column.width = state.width;
```

you should make the `ColumnTSC` type have width be optional, because leyton and i made the `columninfo` variable have optional width so it wouldn't be put in the xml as a tag. right now the width is going to be the default if it's a parent which could be mislead, so i think it should be optional and not be put in the xml if it's not there

---

Since ColumnTSC is supposed to be a direct representation of the settings file, I think making the width property optional on it is not a good idea since every column in a settings file has it even though the column itself might not use it (like the parent columns). I'll set the default to -1 which will clearly indicate that the width hasn't been set for this. The generated settings file from the java program will autofill the width either way. Lmk what you think